### DISCUSSION TITLE
Where do I put hooks?

### DISCUSSION THREAD

**Discussion by Benjamin-Re** [badeball/cypress-cucumber-preprocessor#1170] [Q&A]
I'm trying to use badeball's Before hook from inside /support but i get the following error

`Expected to find a global registry (this usually means you are trying to define steps or hooks in support/e2e.js, which is not supported) (this might be a bug, please report at https://github.com/badeball/cypress-cucumber-preprocessor)`

I tried to put the hook in different support files like index.ts but it is the same error. So where do I put a hook?

For context - I want to get the tags from the feature files to add them to the report. I found that they're all in an environment variable '__cypress_cucumber_preprocessor_dont_use_this_spec' but as the name suggests I shouldn't use it. I suspect that the pickle object of the Before hook could hold that information as well.

**Comment by badeball** [comment]
Hi @Benjamin-Re, you can place hooks in the same place / files as you would place step definitions.

In the future though, I might want to support hooks defined in `support/*`.

**Reply by Benjamin-Re** [reply]
HI and thanks for the reply. I put it in one of the files where the steps are defined but I get the same error. Looks like I have the same issue as this person: https://github.com/badeball/cypress-cucumber-preprocessor/issues/860

**Reply by badeball** [reply]
> Looks like I have the same issue as this person: https://github.com/badeball/cypress-cucumber-preprocessor/issues/860

And what was my reply to them?

**Reply by Benjamin-Re** [reply]
That you need a minimal project to reproduce the error. I don't think I can provide it, sorry. I just found it strange that everything works except for the hooks.
